prefect_gcp/aiplatform.py

```diff
@@ -0,0 +1,358 @@
+    )
+
+    gcp_credentials: GcpCredentials = Field(
+        default_factory=GcpCredentials,
+        description=(
+            "GCP credentials to use when running the configured "
+            "Vertex AI training job."
+        ),
+    )
+    region: str = Field(
+        default=...,
+        description="The region where the Vertex AI custom training job resides.",
+    )
+    image: str = Field(
+        default=...,
+        title="Image Name",
+        description=(
+            "The image to use for a new Vertex AI custom training job. This value must "
+            "refer to an image within either Google Container Registry "
+            "or Google Artifact Registry, like `gcr.io/<project_name>/<repo>/`"
+        ),
+    )
+    env: Dict[str, str] = Field(
+        default_factory=dict,
+        title="Environment Variables",
+        description="Environment variables to be passed to your Cloud Run Job.",
+    )
+    machine_type: str = Field(
+        default="n1-standard-4",
+        description="The machine type to use for the run, which controls the available "
+        "CPU and memory.",
+    )
+    accelerator_type: Optional[str] = Field(
+        default=None, description="The type of accelerator to attach to the machine."
+    )
+    maximum_run_time: datetime.timedelta = Field(
+        default=datetime.timedelta(days=7), description="The maximum job running time."
+    )
+    network: Optional[str] = Field(
+        default=None,
+        description="The full name of the Compute Engine network"
+        "to which the Job should be peered. Private services access must "
+        "already be configured for the network. If left unspecified, the job "
+        "is not peered with any network.",
+    )
+    service_account: Optional[str] = Field(
+        default=None,
+        description=(
+            "Specifies the service account to use "
+            "as the run-as account in Vertex AI. The agent submitting jobs must have "
+            "act-as permission on this run-as account. If unspecified, the AI "
+            "Platform Custom Code Service Agent for the CustomJob's project is "
+            "used. Takes precedence over the service account found in gcp_credentials, "
+            "and required if a service account cannot be detected in gcp_credentials."
+        ),
+    )
+
+    job_watch_poll_interval: float = Field(
+        default=5.0,
+        description=(
+            "The amount of time to wait between GCP API calls while monitoring the "
+            "state of a Vertex AI Job."
+        ),
+    )
+
+    @property
+    def job_name(self):
+        """
+        The name can be up to 128 characters long and can be consist of any UTF-8 characters.
+        Keeps only alphanumeric characters, dashes, underscores, and periods. Reference:
+        https://cloud.google.com/python/docs/reference/aiplatform/latest/google.cloud.aiplatform.CustomJob#google_cloud_aiplatform_CustomJob_display_name
```

Comment:
If this can contain any UTF-8 characters why are we filtering the content?

Reply:
Wasn't sure if some malicious code could be injected so just to be on the safe side. Also consistent with the other infra I think, but I could remove this restriction.

Reply:
Anything that goes into our database should be properly escaped elsewhere. I think we should stick with the restrictions that the backing service has.